sdl_viewer/src/terrain_drawer/layer.rs

```diff
@@ -0,0 +1,232 @@
+use crate::graphic::moving_window_texture::GlMovingWindowTexture;
+use crate::graphic::tiled_texture_loader::TiledTextureLoader;
+use crate::graphic::uniform::GlUniform;
+use crate::graphic::GlProgram;
+use crate::terrain_drawer::read_write::Metadata;
+use cgmath::{Decomposed, Matrix4, Vector2, Vector3, Zero};
+use image::{ImageBuffer, LumaA, Rgba};
+use point_viewer::math::Isometry3;
+use std::convert::TryInto;
+use std::rc::Rc;
+
+const F64_MAX_INT: i64 = 9_007_199_254_740_992;
```

Comment:
nit: Call it `F64_MAX_SAFE_INT`, write it as `2_i64.pow(std::f64::MANTISSA_DIGITS)`, and add a comment explaining what it actually is? (name taken from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER)

Reply:
That's not yet a constant function sadly.

Reply:
ohh, too bad.